File: IronNails.Contracts/Models/TweetBase.cs
using System;

namespace IronNails.Models
{
    public class TweetBase : ModelBase
    {
        private DateTime? _createdAt;
        private string _humanizedTime;

        public string HumanizedTime
        {
            get { return _humanizedTime; }
            set
            {
                if (value == _humanizedTime) return;
                _humanizedTime = value;
                OnPropertyChanged("HumanizedTime");
            }
        }

        public DateTime? CreatedAt
        {
            get { return _createdAt; }
            set
            {
                if (value == _createdAt) return;
                _createdAt = value;
                OnPropertyChanged("CreatedAt");
            }
        }
    }
}

File: IronNails/app/models/tweet_base.rb
require File.dirname(__FILE__) + "/tweet_base"

module IronNails

  module Models

    class TweetBase < ModelBase

      attr_accessor :created_at, :humanized_time

      def humanized_time
        update_humanized_time
      end


      def update_humanized_time
        self.humanized_time = self.created_at.humanize if self.created_at
      end

      def self.collection
        TweetCollection
      end

    end

  end
end
